pointcloud/basic_octree.go

```diff
@@ -1,92 +1,94 @@
-package octree
+package pointcloud
 
 import (
-	"context"
-
-	"github.com/edaniels/golog"
 	"github.com/golang/geo/r3"
 	"github.com/pkg/errors"
-
-	pc "go.viam.com/rdk/pointcloud"
 )
 
 const (
+	internalNode = NodeType(iota)
```

Comment:
Why are these different values being made private as opposed to public? I'm thinking of the usecase of someone using Iterate. Wouldn't they need to be able to know if a given leaf node is an internal node, leaf node, or filled node? If so, wouldn't they need to be able to reference these constants outside of the octree package?

Reply:
Its a minor point but with the other expsoing changes I wanted to put forward the option of have these unexposed as it would limit peoples ability to interact with the octree internals beyond the exposed functions. Internal node vs leaf nodes ca still be denotes by calling size on a specific node (node.Size() == 0 empty leaf, node.Size() == 1 filled leaf, node.Size > 1 internal node. 

However, I'm open to re-exposing them, that was just my thought process behind the change.

Reply:
I'm good with keeping them private initially if you don't think callers will need to access them.